ComponentKit/Core/CKComponentViewConfiguration.mm

```diff
@@ -20,7 +20,12 @@
 
 CKComponentViewClass::CKComponentViewClass(Class viewClass) noexcept :
 identifier(class_getName(viewClass)),
-factory(^{return [[viewClass alloc] init];}) {}
+factory(^{
+  #if DEBUG
+    CKAssertTrue([viewClass isSubclassOfClass:[UIView class]], @"%@ is not a subclass of UIView", viewClass);
+  #endif
+  return [viewClass isSubclassOfClass:[UIView class]] ? [[viewClass alloc] init] : nil;
+}) {}
```

Comment:
no, it is never used as a "precompiled dependency." However, don't bother with #DEBUG, CKAssert is compiled out in non-debug builds for you (in the configuration we use internally)